src/battle-tooltips.ts

```diff
@@ -470,19 +470,26 @@ class BattleTooltips {
 			if (item.zMoveFrom === move.name) {
 				move = this.battle.dex.getMove(item.zMove as string);
 			} else if (move.category === 'Status') {
-				move = new Move(move.id, "", {
-					...move,
-					name: 'Z-' + move.name,
-				});
-				zEffect = this.getStatusZMoveEffect(move);
+				if (item.zMove && item.zMoveType === move.type) {
+					move = new Move(move.id, "", {
+						...move,
+						name: 'Z-' + move.name,
+					});
+					zEffect = this.getStatusZMoveEffect(move);
+				} else {
+					move = this.battle.dex.getMove('maxguard');
+				}
 			} else {
-				let moveid;
+				let moveName;
 				if (item.zMoveType) {
-					moveid = BattleTooltips.zMoveTable[item.zMoveType as TypeName];
+					moveName = BattleTooltips.zMoveTable[item.zMoveType as TypeName];
 				} else {
-					moveid = BattleTooltips.maxMoveTable[move.type as TypeName];
+					moveName = BattleTooltips.maxMoveTable[move.type as TypeName];
+				}
+				if (pokemon.species.includes('Gmax')) {
+					if (toID(move.isMax) === toID(pokemon.species).slice(0, -4)) moveName = move.name;
```

Comment:
This currently only works when the Pokemon has already Gigantamaxed. Any ideas on how to make it work before gigantamaxing?